### ISSUE TITLE
Unable to delegate dependent types

### ISSUE THREAD

**Issue by Atry** [ChrisNeveu/macrame#5]
``` scala
  object Foo {
    def setType(nativeArray: scala.reflect.api.Types#Type): Unit = ???
  }
  object Bar {
    @delegate def foo: Foo.type = Foo
  }
```

```
[error] Bar.scala:20: not found: type Type
[error]     @delegate def foo: Foo.type = Foo
[error]      ^
```

**Comment by ChrisNeveu** [comment]
The macro is pretty naive when it grabs the types for method arguments, just geabbing the type name which is obviously wrong here.

We should be able to check for `#` types and make sure those are transferred correctly.

**Comment by ChrisNeveu** [comment]
Fixed by https://github.com/ChrisNeveu/macrame/commit/f59f494aa7463236dee9bed098b0682e4b9d2dba which is included in the new `1.2.3` release.
